doc/Pacemaker_Explained/en-US/Ch-Rules.txt

```diff
@@ -522,6 +522,121 @@ You may wish to write +end="2005-03-31T23:59:59"+ to avoid confusion.
 -------
 =====
 
+== Resource Expressions ==
+
+An +rsc_expression+ is a rule condition based on a resource agent's properties.
+This rule is only valid within an +rsc_defaults+ or +op_defaults+ context.
+
+.Attributes of an rsc_expression Element
+[width="95%",cols="2m,<5",options="header",align="center"]
+|=========================================================
+
+|Field
+|Description
+
+|id
+|A unique name for the expression (required)
+ indexterm:[XML attribute,id attribute,rsc_expression element]
+ indexterm:[XML element,rsc_expression element,id attribute]
+
+|class
+|The standard name to be matched against resource agents (required)
+ indexterm:[XML attribute,class attribute,rsc_expression element]
+ indexterm:[XML element,rsc_expression element,class attribute]
+
+|provider
+|If given, the vendor to be matched against resource agents.  This
+ only makes sense for agents using the OCF spec.
+ indexterm:[XML attribute,provider attribute,rsc_expression element]
+ indexterm:[XML element,rsc_expression element,provider attribute]
+
+|type
+|The name of the resource agent to be matched (required)
+ indexterm:[XML attribute,type attribute,rsc_expression element]
+ indexterm:[XML element,rsc_expression element,type attribute]
+
+|=========================================================
+
+=== Example Resource-Based Expressions ===
+
+A small sample of how resource-based expressions can be used:
+
+.True for all ocf:heartbeat:IPaddr2 resources
+====
+[source,XML]
+----
+<rule id="rule1" score="INFINITY">
+    <rsc_expression id="rule_expr1" class="ocf" provider="heartbeat" type="IPaddr2"/>
+</rule>
+----
+====
+
+.Provider doesn't apply to non-OCF resources
+====
+[source,XML]
+----
+<rule id="rule2" score="INFINITY">
+    <rsc_expression id="rule_expr2" class="stonith" type="fence_xvm"/>
+</rule>
+----
+====
+
+== Operation Expressions ==
+
+An +op_expression+ is a rule condition based on an action of some resource
+agent.  This rule is only valid within an +op_defaults+ context.
+
+.Attributes of an op_expression Element
+[width="95%",cols="2m,<5",options="header",align="center"]
+|=========================================================
+
+|Field
+|Description
+
+|id
+|A unique name for the expression (required)
+ indexterm:[XML attribute,id attribute,op_expression element]
+ indexterm:[XML element,op_expression element,id attribute]
+
+|name
+|The action name to match against.  This can be any action supported by
+ the resource agent; common values include +monitor+, +start+, and +stop+
+ (required).
+ indexterm:[XML attribute,name attribute,op_expression element]
+ indexterm:[XML element,op_expression element,name attribute]
```

Comment:
OTOH it would be fine to require operation name, it would be more weird to say "all 10-second interval operations of any kind". But it would be just as fine to allow it.

Reply:
Yeah, that 's how it is set up now.  Name is required but interval is optional.  If you give interval, both it and name have to match.  Otherwise just name has to match.